File: DecompiledGameScripts/cyberpunk/devices/masters/actionsSequencer.swift

public class ActionsSequencer extends InteractiveMasterDevice {

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as ActionsSequencerController;
  }

  private const func GetController() -> ref<ActionsSequencerController> {
    return this.m_controller as ActionsSequencerController;
  }

  public const func GetDevicePS() -> ref<ActionsSequencerControllerPS> {
    return this.GetController().GetPS();
  }
}

File: Definitions/Game_Definitions/classes/ActionsSequencer.lua
---@meta
---@diagnostic disable

---@class ActionsSequencer : InteractiveMasterDevice
ActionsSequencer = {}

---@return ActionsSequencer
function ActionsSequencer.new() return end

---@param props table
---@return ActionsSequencer
function ActionsSequencer.new(props) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function ActionsSequencer:OnTakeControl(ri) return end

---@return ActionsSequencerController
function ActionsSequencer:GetController() return end

---@return ActionsSequencerControllerPS
function ActionsSequencer:GetDevicePS() return end

